at_onboarding_cli/lib/src/at_onboarding_service.dart

```diff
@@ -0,0 +1,184 @@
+import 'dart:convert';
+import 'dart:io';
+import 'package:at_client/at_client.dart';
+import 'package:at_commons/at_builders.dart';
+import 'package:at_commons/at_commons.dart';
+import 'package:at_lookup/at_lookup.dart';
+import 'package:at_onboarding_cli/at_onboarding_cli.dart';
+import 'package:at_utils/at_logger.dart';
+import 'package:crypton/crypton.dart';
+import 'package:encrypt/encrypt.dart';
+import 'package:zxing2/qrcode.dart';
+import 'package:image/image.dart';
+
+///class containing service that can onboard/authenticate @signs
+class OnboardingService {
+  String _atSign;
+  bool _isPkamAuthenticated = false;
+  AtLookupImpl? _atLookup;
+  AtClient? _atClient;
+  AtSignLogger logger = AtSignLogger('Onboarding CLI');
+  AtOnboardingConfig atOnboardingConfig;
+
+  AtClient? get atClient => _atClient;
+  AtLookupImpl? get atLookup => _atLookup;
+
+  OnboardingService(this._atSign, this.atOnboardingConfig);
+
+  ///creates instance of [AtClient] using either of [AtClientPreference] or [AtOnboardingConfig]
+  Future<AtClient?> createAtClient() async {
+    AtClientManager _atClientManager = AtClientManager.getInstance();
```

Comment:
* have _isOnBoardedFlag at class level. set the flag to true in onboard method - isCramSuccessful==true
* rename createAtClient to getAtClient.
if(this.atClient !=null) {
return this.atClient
}
if(!_isOnBoardedFlag) {
throw Exception('Cannot get atClient instance without onboarding');
}
//logic to create and return atClient here